libs/test-macros/src/test_each_connector.rs

```diff
@@ -170,22 +155,48 @@ fn test_each_connector_async_wrapper_functions(
         None
     };
 
+    let test_fn_name = &test_function.sig.ident;
+
+    let output = quote! {
+        mod #test_name {
+            #(#tests)*
+
+            fn run(api: std::pin::Pin<Box<dyn std::future::Future<Output = super::TestApi> + 'static + Send>>) {
+                #optional_logging_import
+
+                let fut = async move {
+                    let api = api.await;
+                    super::#test_fn_name(&api).await#optional_unwrap
+                }#optional_logging;
+
+                test_setup::runtime::run_with_tokio(fut);
+            }
+
+        }
+
+        #test_function
+    };
+
+    output.into()
+}
+
+fn test_each_connector_async_wrapper_functions(
```

Comment:
name is wrong now, these functions are no longer async. I'll change it before merging.